wscl-issues/draft/vector-push-type-error

```diff
@@ -0,0 +1,73 @@
+Issue:          VECTOR-PUSH-TYPE-ERROR
+Forum:          Cleanup
+Category:       CLARIFICATION
+Status:         draft
+Edit History:   16-May-24, Version 1 by Jin-Cheng Guu
+References:     VECTOR-PUSH
+
+Problem Description:
+
+  In the draft ANSI Common Lisp specification, the description of the
+  functions VECTOR-PUSH and VECTOR-PUSH-EXTEND indicates no exceptional
+  situations for when the NEW-ELEMENT is not of a type indicated by the
+  ELEMENT-TYPE of the VECTOR.
+
+Proposal (VECTOR-PUSH-TYPE-ERROR:SIGNAL-ERROR-IN-SAFE-CODE):
+
+  This proposal changes the description of the VECTOR-PUSH function, so that
+  the section "Exceptional Situations" instead reads:
+
+  "The functions VECTOR-PUSH and VECTOR-PUSH-EXTEND should signal an error of
+  type TYPE-ERROR if given a NEW-ELEMENT not of a type indicated by the
+  ELEMENT-TYPE of the VECTOR.
+
+Test Cases:
+
+  (defun one ()
+    (declare (optimize (safety 3)))
+    (vector-push 0 (make-array 5 :fill-pointer 2
+                                 :element-type 'character
+                                 :initial-element #\a)))
+
+  (one) => ERROR: The value 5 is not of type CHARACTER.
```

Comment:
Please put the output in a comment and enclose the error in square brackets. We are working on a system that will auto-update the "Current Practices" section based on the "Test Cases" section.

Reply:
Change line 32 to `(one) ; => [signals TYPE-ERROR with expected type CHARACTER]`